src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,722 @@
+            {
+                auto tile = World::TileManager::get(tilePos);
+                auto* surface = tile.surface();
+
+                if (surface->slope() || highestBaseZ != surface->baseZ())
+                {
+                    if (highestBaseZ < surface->baseZ())
+                    {
+                        bool hasPassedSurface = false;
+                        for (auto& el : tile)
+                        {
+                            if (el.type() == World::ElementType::surface)
+                            {
+                                hasPassedSurface = true;
+                                continue;
+                            }
+
+                            if (hasPassedSurface)
+                            {
+                                auto* elTrack = el.as<World::TrackElement>();
+                                auto* elRoad = el.as<World::RoadElement>();
+                                auto* elStation = el.as<World::StationElement>();
+                                auto* elBuilding = el.as<World::BuildingElement>();
+                                auto* elIndustry = el.as<World::IndustryElement>();
+                                auto* elTree = el.as<World::TreeElement>();
+                                if (elTrack != nullptr && !elTrack->isGhost() && elTrack->hasBridge())
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
+                                else if (elRoad != nullptr && !elRoad->isGhost() && elRoad->hasBridge())
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
+                                else if (elStation != nullptr && elStation->unk5SLR5() == 2)
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
+                                else if (elBuilding != nullptr)
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
+                                else if (elIndustry != nullptr)
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
+                                else if (elTree != nullptr && clearZ <= elTree->baseZ())
+                                {
+                                    setErrorText(StringIds::empty);
+                                    return FAILURE;
+                                }
```

Comment:
Bit down to personal taste, but I think something like this would be more readable:
```suggestion
                                auto emptyError = []() { 
                                    setErrorText(StringIds::empty);
                                    return FAILURE;
                                };

                                auto* elTrack = el.as<World::TrackElement>();
                                if (elTrack != nullptr && !elTrack->isGhost() && elTrack->hasBridge())
                                    return emptyError();
                                
                                auto* elRoad = el.as<World::RoadElement>();
                                if (elRoad != nullptr && !elRoad->isGhost() && elRoad->hasBridge())
                                    return emptyError();
                                
                                auto* elStation = el.as<World::StationElement>();
                                if (elStation != nullptr && elStation->unk5SLR5() == 2)
                                    return emptyError();
                                
                                auto* elBuilding = el.as<World::BuildingElement>();
                                if (elBuilding != nullptr)
                                    return emptyError();
                                
                                auto* elIndustry = el.as<World::IndustryElement>();
                                if (elIndustry != nullptr)
                                    return emptyError();
                                
                                auto* elTree = el.as<World::TreeElement>();
                                if (elTree != nullptr && clearZ <= elTree->baseZ())
                                    return emptyError();
```
```